batch/batch/server/deserialize.py

```diff
@@ -0,0 +1,91 @@
+from datetime import date, datetime
+import re
+from kubernetes.client import models
+
+
+def deserialize(api_client, data, klass):  # pylint: disable=R0911
+    if data is None:
+        return None
+
+    if isinstance(klass, str):
+        if klass.startswith('list['):
+            sub_kls = re.match(r'list\[(.*)\]', klass).group(1)
+            return [deserialize(api_client, sub_data, sub_kls)
+                    for sub_data in data]
+
+        if klass.startswith('dict('):
+            sub_kls = re.match(r'dict\(([^,]*), (.*)\)', klass).group(2)
+            return {k: deserialize(api_client, v, sub_kls)
+                    for k, v in data.items()}
+
+        # convert str to class
+        if klass in api_client.NATIVE_TYPES_MAPPING:
+            klass = api_client.NATIVE_TYPES_MAPPING[klass]
+        else:
+            klass = getattr(models, klass)
+
+    if klass in api_client.PRIMITIVE_TYPES:
+        return deserialize_primitive(data, klass)
+    if klass == object:
+        return deserialize_object(data)
+    if klass == date:
+        return deserialize_date(data)
+    if klass == datetime:
+        return deserialize_datatime(data)
+    else:
+        return deserialize_model(api_client, data, klass)
+
+
+def deserialize_primitive(data, klass):
+    try:
+        return klass(data)
+    # except UnicodeEncodeError:
+    #     return unicode(data)  # FIXME: What should this be?
```

Comment:
I believe this should be return `str(data)`

https://www.quora.com/Why-was-unicode-renamed-to-str-in-Python-3